VisualStudioPlugin/NewProjectDialog.xaml.cs

```diff
@@ -0,0 +1,121 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.Linq;
+using System.Windows;
+using System.Windows.Input;
+using System.Windows.Media;
+
+namespace QuantConnect.VisualStudioPlugin
+{
+    /// <summary>
+    /// Interaction logic for NewProjectDialog.xaml
+    /// </summary>
+    internal partial class NewProjectDialog : DialogWindow
+    {
+        /// <summary>
+        /// True if user entered valid data
+        /// </summary>
+        public bool CreateNewProject { get; private set; }
+
+        /// <summary>
+        /// User provided Project name
+        /// </summary>
+        public string NewProjectName { get; private set; }
+
+        /// <summary>
+        /// User provided project programming laguange
+        /// </summary>
+        public Language NewProjectLanguage { get; private set; }
+
+        /// <summary>
+        /// Supported programming languages. This is to filter enum Language, which has unsupported languages.
+        /// </summary>
+        private static string[] _supportedLanguages = { "FSharp", "CSharp", "Python" };
```

Comment:
no change required, but curious why choose to use strings here instead of the `Language` enum directly?
Also, this member can be marked as `readonly`.

Reply:
Your right! Using `Language` would of been a cleaner approach. My brain just wasn't contributing when I thought this through. I'll see to it in the next patch 👌